pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,37 @@
+package pgutils
+
+import (
+	"time"
+)
+
+const (
+	interval = 5 * time.Second
+	timeout  = 5 * time.Minute
+)
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery(fn func() error) error {
```

Comment:
I think it would be nicer to rewrite this as
```suggestion
func RetryQuery(ctx context.Context, fn func(ctx context.Context) error) error {
```
that way, the timeout could be made universal by using a `context.WithTimeout` instead of the expiration timer.

Reply:
Nevermind, since you fatal on timeout, I guess a consistent timeout isn't all that relevant.